lib/dialects/abstract/query-generator.js

```diff
@@ -1917,21 +2117,21 @@ class QueryGenerator {
    * @private
    */
   addLimitAndOffset(options) {
-    let fragment = '';
+    const composition = new Composition();
 
     /* eslint-disable */
```

Comment:
I see now, maybe instead remove the eslint-disable code and use the following instead?

```js
if (options.offset !== undefined && options.offset !== null && (options.limit === null || options.limit === undefined)) {
  // @TODO: Throw an exception or remove this code to circumvent potential DOS vector
  composition.add('LIMIT ', new Slot(options.offset), ', ' + 10000000000000);
} else if (options.limit !== undefined && options.limit !== null) {
  if (options.offset !== undefined && options.offset !== null) {
    composition.add('LIMIT ', new Slot(options.offset), ', ', new Slot(options.limit));
  } else {
    composition.add('LIMIT ', new Slot(options.limit));
  }
}
```

Side note: I think Sequelize should throw an exception when `offset` is passed but not `limit`.

Reply:
I have switched to strict null/undefined comparison.

I agree, Sequelize should at least warn if offset is used without limit. Furthermore, Sequelize should warn if either offset or limit are used without order by. This will produce inconsistent pagination [as warned by postgres](https://www.postgresql.org/docs/current/queries-limit.html) and other DBs.

Reply:
I think a warning is more adequate than an error.

There are some cases in which offset without limit is ok. For instance, let's say your app manages collections of elements such as folders and files, song lists or photo albums. The main page shows a sample of elements from each collection using `SELECT values FROM mytable LIMIT 5`. The user can press the *show all* button to load the rest of the collection without limit using `SELECT values FROM mytable OFFSET 5`.

Perhaps this is anecdote, but sometimes LIMIT clauses are not used for pagination and instead are used for defensive programming. In this case ORDER BY is not required. Let's say your app stores the number of siblings, cars, houses or languages that a person has. A person will typically have around 1 or 2 siblings, cars, houses, languages... They will seldom have 10 or 20. And you know they will never have more than 100. The limit clause protects you from poorly constructed queries(like `id = NULL`) that bring thousands/millions of rows and put the DB to its knees.

Reply:
See https://github.com/sequelize/sequelize/issues/10331